### PR TITLE
Add world restricted attribute to full-screen viewer and disable…

### LINKED ISSUE #488: Users can right click and download an image when in full screen
When viewing a no-download objects in full screen, a user can right click and save image as.  To re-create go to https://purl.stanford.edu/zz234zw6791, expand the image to full screen, right click on the image, choose save image as.  This works in Chrome and Firefox, but not Safari.

### PR DESCRIPTION
**Pull request by jkeck**
… right-clicks.

Fixes #488 

Paired w/ @ndushay on this

cc @mejackreed

### CODE REVIEW

File: app/assets/javascripts/modules/image_x_viewer.js [hunk 1]
```diff
@@ -253,7 +253,10 @@
     var _setupButtonListeners = function() {
       $embedHeader.find('[data-sul-view-fullscreen="fullscreen"]')
         .on('click', function() {
+          var worldRestriction = $('[data-world-restriction]').data('world-restriction');
           canvasStore.osd.setFullScreen(true);
+          $('.osd-container').attr('data-world-restriction', worldRestriction);
```
Comment (jkeck) [inline app/assets/javascripts/modules/image_x_viewer.js]: The `world-restriction` data attribute is not present on the full-screen OSD viewer so we need to add it there so `_disableClicks()` knows to disable the right-click.

File: app/assets/javascripts/modules/image_x_viewer.js [hunk 2]
```diff
@@ -253,7 +253,10 @@
     var _setupButtonListeners = function() {
       $embedHeader.find('[data-sul-view-fullscreen="fullscreen"]')
         .on('click', function() {
+          var worldRestriction = $('[data-world-restriction]').data('world-restriction');
```
Comment (mejackreed) [inline app/assets/javascripts/modules/image_x_viewer.js]: I think you can get away with removing this. This should already be stored in an object `dataAttributes.worldRestriction`.

File: app/assets/javascripts/modules/image_x_viewer.js (hunk 1 above)
Comment (mejackreed) [inline app/assets/javascripts/modules/image_x_viewer.js]: I think you can get away with removing this. This should already be stored in an object `dataAttributes.worldRestriction`.

File: app/assets/javascripts/modules/image_x_viewer.js (hunk 2 above)
Comment (jkeck) [inline app/assets/javascripts/modules/image_x_viewer.js]: :+1: Good call!

I think that was our first attempt at trying to figure out how to disable in full-screen, and when we figured out that we needed to disable the context menus scoped to a different element in full-screen, didn't go back to see if this was ever necessary.

**Comment by mejackreed** [comment]
:+1:

**Comment by ndushay** [comment]
@jkeck let's merge and deploy !

**Comment by peetucket** [comment]
sweet

**Comment by rosy1280** [comment]
Oh sure. Now that I'm gone you close my tickets. Punishment?

**Comment by jkeck** [comment]
@rosy1280, I know!  We have been waiting for you to leave so we could fix all the problems you reported. 😜

**Comment by peetucket** [comment]
Not only closed, but fixed too!

**Comment by ndushay** [comment]
but @peetucket, we didn't deploy it yet.

**Pull request merged**